values.go

```diff
@@ -1229,6 +1229,70 @@ func (v Word64) String() string {
 	return format.Uint(uint64(v))
 }
 
+// Word128
+
+type Word128 struct {
+	Value *big.Int
+}
+
+var _ Value = Word128{}
+
+var Word128MemoryUsage = common.NewCadenceBigIntMemoryUsage(16)
+
+func NewWord128(i uint) Word128 {
+	return Word128{
+		Value: big.NewInt(int64(i)),
+	}
+}
+
+func NewWord128FromBig(i *big.Int) (Word128, error) {
+	if i.Cmp(sema.Word128TypeMaxIntBig) > 0 || i.Sign() < 0 {
+		// When Sign() < 0, Mod will add sema.Word128TypeMaxIntPlusOneBig
+		// to ensure the range is [0, sema.Word128TypeMaxIntPlusOneBig)
+		i.Mod(i, sema.Word128TypeMaxIntPlusOneBig)
```

Comment:
It might be better to just return an error here if the input is out of range, e.g. like in `NewUInt128FromBig`

Reply:
Done